### ISSUE TITLE
`react-aria-components`: `<Checkbox>`'s (hidden) input isn't interactive with mouse

### ISSUE THREAD

**Issue by Refrase** [adobe/react-spectrum#8755]
### Provide a general summary of the issue here

Our Cypress test cases fail when interacting with `react-aria-components` `<Checkbox>`s when targeting them by role as per best practice.
```
cy.findByRole("checkbox")
```
Neither the `.click()` not the `.realClick()` (from [cypress-real-events](https://github.com/dmtrKovalenko/cypress-real-events?tab=readme-ov-file#cyrealclick)) works.

We found it's because the `<input>` element in RAC is 1x1 px and not covering the whole width of checkbox + label as the Spectrum equivalent does.

**RAC Checkbox**
<img width="552" height="263" alt="Image" src="https://github.com/user-attachments/assets/8bda130d-4311-4ef9-9108-9f8c513646c6" />

Also note how body becomes the active element on mousedown for the RAC component:
https://github.com/user-attachments/assets/6add939f-8831-43e7-9853-96803700a1e5

**Spectrum Checkbox**
<img width="727" height="286" alt="Image" src="https://github.com/user-attachments/assets/2e265ba0-93aa-493f-84ff-d6f1c013e9ac" />
<img width="649" height="25" alt="Image" src="https://github.com/user-attachments/assets/a603d342-23c9-4823-b3f1-6598e1816929" />

The `<input>` is always active for the Spectrum component:
https://github.com/user-attachments/assets/3253782e-c78a-48b9-a012-34887ef136ed

Have you considered doing some inline size of the `<input>` element since it's the element coming with the role?
And making that itself the visually hidden component instead of wrapping with `<VisuallyHidden>`.
I imagine accessibility is hurt by having `<body>` element become the active element intermittently on mousedown.

This is btw the case for `<RadioGroup>` and `<Switch>` as well.
It might also be answering the doubts you have regarding HTML validator failing  @devongovett in your PR for making the `<VisuallyHidden>` element a `<span>` (although I guess it accepts the span then):
https://github.com/adobe/react-spectrum/pull/4590

This is our error
```
Timed out retrying after 4050ms: `cy.click()` failed because this element:  `<input type="checkbox" data-react-aria-pressable="true" tabindex="0" value="Routing">`  is being covered by another element:  `<span class="KFl+5MGj5Rrh+DThGrQnqg=="></span>`...
```

### 🤔 Expected Behavior?

RAC `<Checkbox>`'s `<input>` element should also be active on mousedown.

### 😯 Current Behavior

`<body>` element is intermittently active on mousedown on the checkbox box.

### 💁 Possible Solution

Increasing inline-size of the `<input>` element to cover box + label and make the `<input>` itself visually hidden (without wrapping in span).

### 🖥️ Steps to Reproduce

https://react-spectrum.adobe.com/react-spectrum/Checkbox.html#example
https://react-spectrum.adobe.com/react-aria/Checkbox.html#example

### Version

Not using it

### What browsers are you seeing the problem on?

Other

### If other, please specify.

Electron 106

### What operating system are you using?

Linux

**Comment by snowystinger** [comment]
Looks like Cypress documentation recommends `check` and `uncheck`? https://docs.cypress.io/api/commands/check
More accurate to a real user though, you can get the label and click that though I'm pretty sure.

**Comment by Refrase** [comment]
Seems that has the same issue:
<img width="575" height="392" alt="Image" src="https://github.com/user-attachments/assets/ed3faaa6-1c28-41a8-9d3e-92d51673d010" />

**Comment by snowystinger** [comment]
Have you tried using `{ force: true }`? it sounds like that is the recommended way to do it for a visually hidden checkbox

**Comment by snowystinger** [comment]
closing due to inactivity
